src/graphql/resolvers.ts

```diff
@@ -23,41 +29,37 @@ export const resolvers = {
       hasLettersAndNumbers(data.password);
       isValidDate(data.birthDate);
       isValidYear(data.birthDate);
+      await isValidEmail(data.email);
 
       const hashedPassword = await hashPassword(data.password);
 
-      try {
-        const newUser = await prisma.user.create({
-          data: {
-            name: data.name,
-            password: hashedPassword,
-            email: data.email,
-            birthDate: data.birthDate
-          }
-        });
-
-        return newUser;
-      } catch (error) {
-        if (error instanceof Prisma.PrismaClientKnownRequestError) {
-          await invalidEmail(data.email);
+      const newUser = await prisma.user.create({
+        data: {
```

Comment:
```suggestion
      return prisma.user.create({
        data: {
```

Esse atalho é possível, dado que vc só tá retornando o valor resolvido dessa Promise, então vc pode retornar a promise direto. É como se vc criasse uma variável local e retornasse logo em seguida , mesmo princípio:

```ts
function soma(a, b) {
  const sum = a + b;
  return sum;
}

// ------------
function soma(a, b) {
  return a + b;
}
```